snapshotter/core_api.py

```diff
@@ -131,6 +157,36 @@ async def health_check(
                 'status': 'error',
                 'message': 'Snapshotter is not active',
             }
+
+    # check if there are any unfinalized projects every 10 minutes
+    if int(time.time()) - app.state.last_unfinalized_check >= 600:
+        app.state.last_unfinalized_check = int(time.time())
+        unfinalized_projects: List[UnfinalizedProject] = await snapshotter_last_finalized_check(
+            redis_conn,
+            request.app.state.protocol_state_contract,
+            request.app.state.anchor_rpc_helper,
+        )
+
+        if unfinalized_projects:
+            for unfinalized_project in unfinalized_projects:
```

Comment:
Everything looks good! My only concern is that if the node is down for some reason, since `core_api` is independent of that, it will keep blasting a lot of failure notifications to Slack, hitting the rate limit. Let's limit the number of failure notifications we send out per minute.

@anomit,  what do you think?

Reply:
yes memoize this to have a check on the outflow of notifications

Reply:
I have added a check to see if an unfinalized project has already been reported to limit the number of notifications in the case where projects cannot finalize. The endpoint will only report a maximum of 5 projects per node every 10 minutes, and if the network goes down, they will only report as many projects that have been assigned to each node.

[cc701ff](https://github.com/PowerLoom/pooler/commit/cc701ffee49d2ebdd8e69904a8e0a1292a54a643)